test/helpers/kubectl.go

```diff
@@ -2484,14 +2484,14 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
 		}
 	}
 
-	if RunsOn419OrLaterKernel() {
+	if RunsOnNetNextKernel() {
 		// To enable SA for both cases when KPR is enabled and disabled
 		addIfNotOverwritten(options, "sessionAffinity", "true")
 	}
 
 	// Disable unsupported features that will just generated unnecessary
 	// warnings otherwise.
```

Comment:
Seems like now the decision isn't about available support, but actually about how we're choosing to configure Cilium depending on the environment:

```suggestion
	// Disable unsupported features that will just generated unnecessary
	// warnings otherwise.
```

Reply:
Also, the change makes this condition and the next conditional check exactly the same so we can combine the two. Same for the conditional check immediately above.

Reply:
> how we're choosing to configure Cilium depending on the environment

I think that was always the case for the e2e ginkgo tests. The configuration was implicitly based on mostly `KERNEL` and `KUBEPROXY` env vars.

Reply:
I messed up the code suggestion upon first try above, I was just pointing out general code cleanups to make this easier to read. I see what you mean about the right path forward being just to drop the code entirely instead of spending more time iterating on nits like this. Makes sense :+1: